src/MessageList/MessageListItem.vala

```diff
@@ -353,13 +353,26 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         if (message_is_html) {
             web_view.load_html (message_content);
         } else {
-            web_view.load_plain_text (message_content);
+            /*
+             * Instead of calling web_view.load_plain_text, use Camel's ToHTML
+             * filter to convert text to HTML. This gives us some niceties like
+             * clickable URLs and email addresses for free.
+             *
+             * Explanation of MimeFilterToHTMLFlags:
+             * https://wiki.gnome.org/Apps/Evolution/Camel.MimeFilter#Camel.MimeFilterToHtml
+             */
+            var flags = Camel.MimeFilterToHTMLFlags.CONVERT_NL |
+                Camel.MimeFilterToHTMLFlags.CONVERT_SPACES |
+                Camel.MimeFilterToHTMLFlags.CONVERT_URLS |
+                Camel.MimeFilterToHTMLFlags.CONVERT_ADDRESSES;
+            var html = Camel.text_to_html (message_content, flags, 0);
+            web_view.load_html (html);
         }
     }
 
-    private async void parse_mime_content (Camel.DataWrapper message_content) {
-        if (message_content is Camel.Multipart) {
-            var content = message_content as Camel.Multipart;
+    private async void parse_mime_content (Camel.DataWrapper mime_content) {
```

Comment:
I've renamed this because having both an instance variable and a parameter called `message_content` is a bit confusing